sdk/servicebus/service-bus/src/receiver.ts

```diff
@@ -146,6 +153,10 @@ export class Receiver {
         return;
       })
       .catch((err) => {
+        if (this._context.streamingReceiver != null && !this._context.streamingReceiver.isOpen()) {
+          this._context.streamingReceiver = undefined;
+        }
```

Comment:
What scenario does this cover?

Reply:
This is the "init() failed and we have a streaming receiver that is not open/doesn't own any resources".

There was a test covering this (`Streaming - Failed init should not cache recevier`) that revealed this behavior which seems sensible. If a streaming receiver is just completely dead (ie, nothing to clean up) it's safe to just not cache it at all.

Reply:
So, this change is because we are now caching first, initializing later?

A cached streaming receiver symbolizes a receiver that needs to be recovered when there is a connection issue. I am slightly concerned on the implications of this change in order on link recovery.

Take the case of init() failing and a connection issue happening at around the same time.
If the connection recovery parts of the code get executed before the catch block here, then this streaming receiver would be on its merry way to being recovered leading to something like #5541

Can we refactor so that we keep the init first, cache later as before, but still have the changes you want?

```js
const sReceiver = StreamingReceiver.create(..);
sReceiver.init().then(() => {
      if (this.isClosed) {
         await sReceiver.close();
         return;
      };
      this._context.streamingReceiver = sReceiver;
      sReceiver.receive(...)
  }).catch (err) { 
  onError(err); 
  }
```

Also, related issue and PR from the past: #1730 and #2139